backend/de.metas.manufacturing/src/main/java/org/eevolution/productioncandidate/material/planning/PPOrderCandidateAdvisedEventCreator.java

```diff
@@ -207,17 +208,13 @@ public BigDecimal handleQuantityDecrease(final @NonNull SupplyRequiredDecreasedE
 		final Candidate demandCandidate = candidateRepositoryRetrieval.retrieveById(CandidateId.ofRepoId(event.getSupplyRequiredDescriptor().getDemandCandidateId()));
 		return candidateRepositoryWriteService.getSupplyCandidatesForDemand(demandCandidate, CandidateBusinessCase.PRODUCTION)
 				.stream()
-				.map(candidate -> ProductionDetail.cast(candidate.getBusinessCaseDetail()).getPpOrderCandidateId())
+				.map(candidate -> ProductionDetail.castOrNull(candidate.getBusinessCaseDetail()).getPpOrderCandidateId())
```

Comment:
better extract the lambda as a static method. handle the null case.

Reply:
done